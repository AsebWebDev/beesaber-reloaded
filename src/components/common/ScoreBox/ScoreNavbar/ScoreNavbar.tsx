import {
  MDBNavbar,
  MDBNavbarItem,
  MDBNavbarLink,
  MDBSwitch,
} from 'mdb-react-ui-kit';
import styled from 'styled-components';

import tokens from '@/tokens';

import type { NavTabs } from '../ScoreBox';

const NavBar = styled(MDBNavbar)`
  background-color: ${tokens.color.page.bgColor.light};
  list-style: none;
  padding: 0 1rem;

  .nav-link {
    color: ${tokens.color.white.light};
  }

  .active {
    background-color: ${tokens.color.white.light};
    color: ${tokens.color.page.bgColor.main};
  }
`;

const NavigationItems = styled.div`
  display: flex;
`;

type Props = {
  activeitem: NavTabs;
  setIsPlayedByHive: (isOn: boolean) => void;
  toggleTab: (id: NavTabs) => void;
};

const ScoreNavbar = ({
  activeitem,
  setIsPlayedByHive,
  toggleTab,
}: Props): JSX.Element => (
  <NavBar>
    <NavigationItems>
      <MDBNavbarItem>
        <MDBNavbarLink
          to="#"
          active={activeitem === 'RECENT'}
          onClick={() => toggleTab('RECENT')}
          role="tab"
        >
          Recent
        </MDBNavbarLink>
      </MDBNavbarItem>
      <MDBNavbarItem>
        <MDBNavbarLink
          to="#"
          active={activeitem === 'TOP'}
          onClick={() => toggleTab('TOP')}
          role="tab"
        >
          Top
        </MDBNavbarLink>
      </MDBNavbarItem>
    </NavigationItems>
    <MDBNavbarItem>
      <MDBSwitch
        label="songs only shared by hive"
        onChange={(e: { target: { checked: boolean } }) =>
          setIsPlayedByHive(e.target.checked)
        }
      />
    </MDBNavbarItem>
  </NavBar>
);

export default ScoreNavbar;
