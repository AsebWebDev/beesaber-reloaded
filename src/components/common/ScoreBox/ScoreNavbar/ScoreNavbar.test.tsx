import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import ScoreNavbar from './ScoreNavbar';

import type { NavTabs } from '../ScoreBox';

describe('ScoreNavbar', () => {
  it.each(['RECENT', 'TOP'] as const)(
    'should match the snapshot when activeItem is %s',
    (activeItem) => {
      const { container } = render(
        <ScoreNavbar
          activeItem={activeItem}
          setIsPlayedByHive={jest.fn()}
          toggleTab={jest.fn()}
        />
      );

      expect(container.firstChild).toMatchSnapshot();
    }
  );

  it.each`
    name        | activeItem
    ${'Recent'} | ${'RECENT'}
    ${'Top'}    | ${'TOP'}
  `(
    'should call toggleTab with $activeItem when $name is clicked',
    ({ activeItem, name }: { activeItem: NavTabs; name: string }) => {
      const mockedToggleTab = jest.fn();

      render(
        <ScoreNavbar
          activeItem={activeItem}
          setIsPlayedByHive={jest.fn()}
          toggleTab={mockedToggleTab}
        />
      );

      const item = screen.getByRole('tab', { name });

      userEvent.click(item);
      expect(mockedToggleTab).toHaveBeenCalledTimes(1);
      expect(mockedToggleTab).toHaveBeenCalledWith(activeItem);
    }
  );

  it('should call setIsPlayedByHive when toggle is clicked', () => {
    const mockedSetIsPlayedByHive = jest.fn();

    render(
      <ScoreNavbar
        activeItem={'RECENT'}
        setIsPlayedByHive={mockedSetIsPlayedByHive}
        toggleTab={jest.fn()}
      />
    );

    const toggle = screen.getByRole('checkbox');

    userEvent.click(toggle);
    expect(mockedSetIsPlayedByHive).toHaveBeenCalledTimes(1);
  });
});
