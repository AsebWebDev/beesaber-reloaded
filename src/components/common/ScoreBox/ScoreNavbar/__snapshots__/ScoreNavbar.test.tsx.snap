// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreNavbar should match the snapshot when activeItem is RECENT 1`] = `
.c0 {
  background-color: #162530;
  list-style: none;
  padding: 0 1rem;
}

.c0 .nav-link {
  color: rgba(255,255,255,0.7);
}

.c0 .active {
  background-color: rgba(255,255,255,0.7);
  color: #00111f;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<nav
  class="navbar c0"
  role="navigation"
>
  <div
    class="c1"
  >
    <li
      class="nav-item"
    >
      <a
        class="nav-link active"
        data-test="nav-link"
        role="tab"
        to="#"
      >
        Recent
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        data-test="nav-link"
        role="tab"
        to="#"
      >
        Top
      </a>
    </li>
  </div>
  <li
    class="nav-item"
  >
    <div
      class="form-check form-switch"
    >
      <input
        class="form-check-input"
        type="checkbox"
      />
      <label
        class="form-check-label"
      >
        songs only shared by hive
      </label>
    </div>
  </li>
</nav>
`;

exports[`ScoreNavbar should match the snapshot when activeItem is TOP 1`] = `
.c0 {
  background-color: #162530;
  list-style: none;
  padding: 0 1rem;
}

.c0 .nav-link {
  color: rgba(255,255,255,0.7);
}

.c0 .active {
  background-color: rgba(255,255,255,0.7);
  color: #00111f;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<nav
  class="navbar c0"
  role="navigation"
>
  <div
    class="c1"
  >
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        data-test="nav-link"
        role="tab"
        to="#"
      >
        Recent
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="nav-link active"
        data-test="nav-link"
        role="tab"
        to="#"
      >
        Top
      </a>
    </li>
  </div>
  <li
    class="nav-item"
  >
    <div
      class="form-check form-switch"
    >
      <input
        class="form-check-input"
        type="checkbox"
      />
      <label
        class="form-check-label"
      >
        songs only shared by hive
      </label>
    </div>
  </li>
</nav>
`;
